### PR TITLE
Fix histogram type = random for nbins = 2

### PR DESCRIPTION
**Pull request by michalkurka**
There is also an issue that the level of freedom is always nbins-1
as opposed to nbins, this is a minor issue in most settings
but it can theoretically change results (=trees will look different).

### CODE REVIEW

File: h2o-algos/src/main/java/hex/tree/GlobalQuantilesCalc.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import hex.quantile.Quantile;
+import hex.quantile.QuantileModel;
+import water.DKV;
+import water.Job;
+import water.Key;
+import water.fvec.Frame;
+import water.util.ArrayUtils;
+
+public class GlobalQuantilesCalc {
```
Comment (maurever) [inline h2o-algos/src/main/java/hex/tree/GlobalQuantilesCalc.java:11]: :+1: 

Could you add some short description to the docstring about usage, please?

File: h2o-algos/src/test/java/hex/tree/DHistogramTest.java [hunk 2]
```diff
@@ -71,4 +71,48 @@ public void findBinForNegativeZero() {
...
+      final double[] values = new double[101];
+      final double step = (maxEx - min) / (values.length - 1);
+      values[0] = min;        
+      for (int i = 1; i < values.length; i++) {
+        values[i] = values[i - 1] + step;
+      }
+      assertEquals(maxEx, values[values.length - 1], 1e-8);
+      values[values.length - 1] = maxEx - 1e-8;
+
+      DHistogram histo_rand = new DHistogram("rand", 20, 1024, (byte) 0, min, maxEx, false, -0.001,
```
Comment (maurever) [inline h2o-algos/src/test/java/hex/tree/DHistogramTest.java:89]:
```suggestion
      DHistogram histoRand = new DHistogram("rand", 20, 1024, (byte) 0, min, maxEx, false, -0.001,
```

File: h2o-algos/src/test/java/hex/tree/DHistogramTest.java [hunk 3]
```diff
@@ -71,4 +71,48 @@ public void findBinForNegativeZero() {
...
+              SharedTreeModel.SharedTreeParameters.HistogramType.Random, 42L, null, null);
+      histo_rand.init();
+
+      // project the random split points into regular space (original values of the column)
+      double[] splitPoints_quant = new double[histo_rand._splitPts.length];
```
Comment (maurever) [inline h2o-algos/src/test/java/hex/tree/DHistogramTest.java:94]:
```suggestion
      double[] splitPointsQuant = new double[histo_rand._splitPts.length];
```

**Review by maurever — CHANGES_REQUESTED** [review]
Looks good to me, just few comments. 

Some tests failing, is it related?

File: h2o-algos/src/test/java/hex/rulefit/RuleFitUtilsTest.java [hunk 4]
```diff
@@ -144,24 +148,19 @@ public void extractRulesFromTreeFromModelTesCategorical() {
             Set<Rule> treeRules =  Rule.extractRulesFromTree(sharedTreeSubgraph, 0);
             assertEquals(treeRules.size(), 8);
 
-            Condition condition1 = new Condition(0, Condition.Type.Categorical, Condition.Operator.In, -1.0, new String[] {"f1995", "f1996", "f1997", "f1998", "f1999", "f2000"}, new int[] {8, 9, 10, 11, 12, 13}, "fYear", false);
```
Comment (michalkurka) [inline h2o-algos/src/test/java/hex/rulefit/RuleFitUtilsTest.java:147]: @koniecsveta I changed the expectations - I found that the natural language representation is easier to work with, please let me know if you see an issue

Comment (koniecsveta) [reply h2o-algos/src/test/java/hex/rulefit/RuleFitUtilsTest.java:147]: LGTM

File: h2o-algos/src/main/java/hex/tree/GlobalQuantilesCalc.java (hunk 1 above)
Comment (michalkurka) [reply h2o-algos/src/main/java/hex/tree/GlobalQuantilesCalc.java]: done, thank you for reminding me

**Comment by michalkurka** [comment]
@maurever implemented your suggestions - thank you

**Review by maurever — APPROVED** [review]
LGTM, thanks @michalkurka.

**Pull request merged**